snow/validators/set.go

```diff
@@ -137,13 +64,6 @@ func (s *vdrSet) add(nodeID ids.NodeID, pk *bls.PublicKey, txID ids.ID, weight u
 		return errDuplicateValidator
 	}
 
-	// We first calculate the new total weight of the set, as this guarantees
```

Comment:
I would suggest to keep the error checking as it is.
I mean check that no overflows happen when inserting/updating weight and keep the query methods like TotalWeight, error free.
This way we are guarantee to fail as fast as possible and to have the set object in a state where all weights are always valid.

Reply:
This is one of the major goals of this PR. So that the node can safely track all subnets without erroring.